internal/runner/accept.go

```diff
@@ -114,10 +111,37 @@ func (r *Runner) accept(ctx context.Context, id string) error {
 	// Open a new job stream. NOTE: we purposely do NOT use ctx above
 	// since if the context is cancelled we want to continue reporting
 	// errors.
-	log.Debug("opening job stream")
-	client, err := r.client.RunnerJobStream(streamCtx)
-	if err != nil {
-		return err
+	retry := false
+	var client pb.Waypoint_RunnerJobStreamClient
+	for {
```

Comment:
Does this need any kind of timeout awareness to prevent infinite loop retries? Maybe infinite loop retries is by design!